### PR TITLE
fix(forms): transform dot-notated errors to nested objects

### LINKED ISSUE #71: Form Errors Reactivity & Nested Fields Errors
### Describe the bug

There is def. something weird with the errors and im not sure yet why, but ill leave this issue to follow up

- The first i noticed while investigating some time ago, is that errors behave in a weird way in hybridly ( and a bit different from inertia ), when the backend sends the same error twice for a specific field, the errors variable is not triggered again ( and makes sense since the message is the same, and so the value never changes, so it will not trigger the watch effect. I believe this is related to original issue, reason why i wanted to mention.

- Seems like when using nested fields in forms, if we submit and the field has an error, the error is correctly set and all good here, but if you attempt to fill the said field ( so it does not contain errors ), the form.errors for that specific field seems like its never removed, while the server sends back a response without the given error shared. Its expected that also nested fields are also cleared when they are not present.

- Please do note that is also breaks the watch effect, so if you have a v-if for the given error, it will always be shown, because the variable is never removed.
 
 The only way to proper workaround this is to use the before hook to clear the given fields, but i believe this should be looked properly to have a smooth expirience.
 
 
 To note also that nested fields errors must be accessed with `form.errors['foo.nested']` instead of dot notation, ( not 100% sure but i believe it would be a nice DX to provide nested navigation for those as well.
 

### Reproduction

https://github.com/nikuscs/hybridly-demo-validation-errors

### Steps to reproduce

- Visit the https://blue-bird.test/login
- First submit without the nested field so it the error is flashed
- Then fill the contents of nested field and send again
- Errors for the given field will not be sent back by the server, but still persisted in errors
- Watch console for proper errors.

### System information

```bash
System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M2
    Memory: 201.41 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.8.0 - ~/Library/Caches/fnm_multishells/68334_1681237407108/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.2.0 - ~/Library/Caches/fnm_multishells/68334_1681237407108/bin/npm
  Browsers:
    Brave Browser: 112.1.50.114
    Chrome: 111.0.5563.146
    Firefox: 110.0
    Safari: 16.4
```

### Used package manager

pnpm

### Logs

_No response_

### Validations

- [X] Read the [docs](https://hybridly.dev).
- [X] Check that there isn't [already an issue](https://github.com/hybridly/hybridly/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Hybridly issue and not an issue related to something else (Vite, Vue...). For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A, join our [Discord server](https://discord.gg/uZ8eC7kRFV).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by nhedger**
This PR is an attempt at fixing #71.

---

This PR updates the `setErrors`, `clearError` and `clearErrors` functions exposed by the `useForm` composable.

### `setErrors`

`setErrors` now expands dot-notated error keys and places the error message in an object structure that follows the structure of the form's fields. This change does not impact simple forms that do not use nesting, and they'll continue working as is.

#### Current behaviour
Assuming that all fields of the following form are required.

```ts
const form = useForm({
    url: '/test',
    method: 'POST',
    fields: {
        name: '',
        phone: {
            work: ''
        }
    }
})
```

The errors returned by Laravel would be:

```json
{
  "name": "The name field is required.",
  "phone.work": "The phone.work field is required."
}
```

and the error object tracked by Hybridly would be:

```json
{
  "name": "The name field is required.",
  "phone.work": "The phone.work field is required."
}
```

#### New behaviour

`setErrors` expands the dot-notated keys and sets the error message in the correct place.

The errors returned by Laravel are the same:

```json
{
  "name": "The name field is required.",
  "phone.work": "The phone.work field is required."
}
```

But the errors tracked by Hybridly become:

```json
{
  "name": "The name field is required.",
  "phone": {
    "work": "The phone.work field is required."
  }
}
```

### `clearError`

`clearError` now accepts a dot-notated path to the error to clear.

```ts
const form = useForm({
    url: '/test',
    method: 'POST',
    fields: {
        name: '',
        phone: {
            work: ''
        }
    }
})

form.clearError('phone.work')
```

When clearing an error, if the parent object is left empty (no more errors), the parent object is also cleared. In this example, the `phone` object will be cleared if the `work` error has been cleared.

---

Two new `setValueAtPath` and `unsetPropertyAtPath` utilities have been created and tests have been created for them.

### CODE REVIEW

File: packages/vue/src/composables/form.ts [hunk 1]
```diff
@@ -177,7 +177,7 @@ export function useForm<T extends Fields = Fields>(options: FormOptions<T>) {
 	/**
 	 * Clears all errors.
 	 */
-	function clearErrors(...keys: (keyof T)[]) {
+	function clearErrors(...keys: string[]) {
```
Comment (innocenzi) [inline packages/vue/src/composables/form.ts]: Do you think we can have auto-completion there using `@clickbar/dot-diver` like [`useProperty`](https://github.com/hybridly/hybridly/blob/0.x/packages/vue/src/composables/property.ts)?

**Review by innocenzi — COMMENTED** [review]
That's really cool, thank you!

File: packages/vue/src/composables/form.ts (hunk 1 above)
Comment (nhedger) [reply packages/vue/src/composables/form.ts]: I'll have a look, but I agree it would be neat.

Comment (nhedger) [reply packages/vue/src/composables/form.ts]: This is proving harder than expected. 

Wondering if we could merge this as-is and open another PR for the auto-completion.

Comment (innocenzi) [reply packages/vue/src/composables/form.ts]: I'll take a look myself tomorrow and see if we can merge then. 👍

Comment (nhedger) [reply packages/vue/src/composables/form.ts]: Sounds good, thanks!

**Comment by innocenzi** [comment]
I fixed all the typings. Thank you for this PR!

**Comment by nhedger** [comment]
Awesome, thank you @innocenzi !

**Pull request merged**